manifests/configuration.pp

```diff
@@ -0,0 +1,110 @@
+# Manage sysctl variable values.
+#
+# Parameters:
+#  $value:
+#    The value for the sysctl parameter. Mandatory, unless $ensure is 'absent'.
+#  $prefix:
+#    Optional prefix for the sysctl.d file to be created. Default: none.
+#  $ensure:
+#    Whether the variable's value should be 'present' or 'absent'.
+#    Defaults to 'present'.
+#
+# Sample Usage :
+#  sysctl::configuration { 'net.ipv6.bindv6only': value => '1' }
+define sysctl::configuration (
+  String $variable               = $title,
+  String $ensure                 = 'present',
+  Variant[String,Undef] $value   = undef,
+  Variant[String,Undef] $prefix  = undef,
+  String $suffix                 = '.conf',
+  Variant[String,Undef] $comment = undef,
+  Variant[String,Undef] $content = undef,
+  Variant[String,Undef] $source  = undef,
+  Boolean $enforce               = true,
+  String $sysctl_binary          = '/sbin/sysctl',
+  String $sysctl_dir_path        = '/etc/sysctl.d',
+) {
+
+  # If we have a prefix, then add the dash to it
+  if $prefix {
+    $_sysctl_d_file = "${prefix}-${variable}${suffix}"
+  } else {
+    $_sysctl_d_file = "${variable}${suffix}"
```

Comment:
No, I don't think they should. `variable` is what I intend to be used, and it takes on `title` as its default value. And that is working just fine in your "basic usage" example. You don't quote the errors - what are they?